### PR TITLE
Update to Photon RC3

### PR DESCRIPTION
**Pull request by briandealwis**
- pulls out common dependencies into `eclipse/common-dependencies.tpd`
- uses latest SWTBot to ensure consistency across all supported platforms

### CODE REVIEW

File: eclipse/common-dependencies.tpd [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  org.slf4j.apis.log4j
+  org.slf4j.bridge.jul
+  ch.qos.logback.slf4j
+}
+
+location "http://dadacoalition.org/yedit/" {
+  org.dadacoalition.yedit.feature.feature.group
+}
+
+location "http://download.eclipse.org/technology/swtbot/milestones/photon-m7/" {
```
Comment (chanseokoh) [inline eclipse/common-dependencies.tpd]: Maybe a comment why we have a Photon-specific location in the common file.

File: eclipse/common-dependencies.tpd [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Common Dependencies for the different platform Target Platform Dependencies.
+ */
+target "Cloud Tools for Eclipse: Common Dependencies"
```
Comment (chanseokoh) [inline eclipse/common-dependencies.tpd]: Is the file will be included as-is? If so, is it OK to have multiple `target`s defined?

Comment (briandealwis) [reply eclipse/common-dependencies.tpd]: Yeah it's fine.  The .target files have all been generated from this.  I have another tweak coming to simplify the generation.

File: pom.xml [hunk 3]
```diff
@@ -33,6 +33,7 @@
     <module>features</module>
     <module>plugins</module>
     <module>third_party</module>
+    <module>eclipse</module>
```
Comment (chanseokoh) [inline pom.xml]: So the difference will be that now we build `eclipse/neon`, `eclipse/oxygen`, and `eclipse/photon` by default at the top level (of course except `eclipse/ide-target-platform`)?

File: pom.xml [hunk 4]
```diff
@@ -479,9 +474,6 @@
          <jettyMaxVersion>9.5</jettyMaxVersion>
       </properties>
       <!-- build against a known target platform -->
```
Comment (chanseokoh) [inline pom.xml]: Remove the comment together (also for the other two profiles).

That said, I wonder how these profiles will know and select which target platform to use.

File: pom.xml (hunk 3 above)
Comment (briandealwis) [reply pom.xml]: We were already pulling in these three projects when importing with our previous instructions.  They will still be found and selected by default under the new instructions, though we will now say to unselect them.

File: pom.xml (hunk 4 above)
Comment (briandealwis) [reply pom.xml]: That's done in the the next set of line for each of those, and why the comment is still relevant.

File: pom.xml (hunk 3 above)
Comment (briandealwis) [reply pom.xml]: I should note that pulling in `eclipse/{neon,oxygen,photon}` by default adds a negligible overhead.  It's `eclipse/ide-target-platform` that takes a while:
```
[INFO] gcp-eclipse-neon 4.6.0-SNAPSHOT .................... SUCCESS [  0.003 s]
[INFO] gcp-eclipse-oxygen 4.7.0-SNAPSHOT .................. SUCCESS [  0.002 s]
[INFO] gcp-eclipse-photon 4.8.0-SNAPSHOT .................. SUCCESS [  0.002 s]
[INFO] ide-target-platform.repo ........................... SUCCESS [01:02 min]
[INFO] eclipse-setup 0.1.0-SNAPSHOT ....................... SUCCESS [  0.002 s]
```

**Comment by briandealwis** [comment]
Ideally I'd like to have a way to tell M2Eclipse to not import the subprojects of `eclipse`, but I wasn't able to find any method with a quick search.

**Pull request merged**